hr_switzerland/models/hr_attendance.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2018 Compassion CH (http://www.compassion.ch)
+#    @author: Stephane Eicher <seicher@compassion.ch>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+
+from odoo import models, api
+
+
+class HrAttendance(models.Model):
+    _inherit = 'hr.attendance'
+
+    ##########################################################################
+    #                               ORM METHODS                              #
+    ##########################################################################
+    @api.model
+    def create(self, vals):
+        """ Log the agent related to the user during the checking """
+        new_record = super(HrAttendance, self).create(vals)
+        user = new_record.employee_id.user_id
+        if 'check_in' in vals and user.resource:
+            user.asterisk_connect(True)
+            self.env['hr.asterisk.autolog'].agent_connexion(ast_user=user,
+                                                            connect=True)
+
+        return new_record
+
+    @api.multi
+    def write(self, vals):
+        """ Logout the agent related to the user during the checking """
+        super(HrAttendance, self).write(vals)
+        user = self.employee_id.user_id
```

Comment:
Here it will fail if self contains multiple record (although you test it just in the next line). I would suggest to add a for loop to respect the api.multi of the write method. `for attendance in self:`